dataplane/api-server/src/netutils.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::Error;
+use libc::if_nametoindex as libc_if_nametoindex;
+use regex::Regex;
+use std::ffi::CString;
+use std::net::Ipv4Addr;
+use std::process::{Command, Stdio};
+use std::str::from_utf8;
+
+/// Returns an ifindex for a provided ifname. Wraps libc.
+pub fn if_nametoindex(ifname: String) -> Result<u32, Error> {
+    let ifname_c = CString::new(ifname)?;
+    let ifindex = unsafe { libc_if_nametoindex(ifname_c.into_raw()) };
+    Ok(ifindex)
+}
+
+/// Given an IPv4 address will return the local system's network interface
+/// which is responsible for routing that address. Not portable: only works on
+/// Linux systems with iproute2 installed.
+///
+/// TODO: replace this https://github.com/Kong/blixt/issues/49
+pub fn if_name_for_routing_ip(ip_addr: Ipv4Addr) -> Result<String, Error> {
```

Comment:
In the future we may be able to interact with netlink similar to what's done here https://github.com/redhat-et/emerita/blob/ade6ca9c1215704da6ce033e0b4e8a5f39dbd908/emerita/src/netlink.rs#L46 just using rtnetlink instead

Reply:
added a note about this to https://github.com/Kong/blixt/issues/49